docs/ASDK-2.0-Layout-Documention.md

```diff
@@ -0,0 +1,802 @@
+    * [calculateSizeThatFits and <code>layout</code>](#calculatesizethatfits-and-layout)
+    * [Example](#example)
+* [Ressources:](#ressources)
+
+## Introduction
+// TODO: Small introduction with an example
+
+## Differences from web
+The goal of AsyncDisplayKit's layout system is not to re-implement all of css. It only targets a subset of css and flexbox kind container, and does not have any plans on implementing support for tables, floats, or any other css concepts. The layout system itself also does not plan on supporting styling properties which do not affect layout such as color or background properties.
+
+The layout system tries to stay as close as possible to css. There are however certain cases where it differs from the web:
+
+### Naming properties
+Certain properties have a different naming as on the web. For example `min-height` equivalent is the `minHeight` property. The full list of properties that control layout is documented in the `Layout Properties` section.
+
+### No margin / padding properties
+Layoutables don't have a padding or margin property. Instead wrapping a layoutable within a `ASInsetLayoutSpec` to apply padding or margin to the layoutable is the recommended way. See `ASInsetLayout` section for more information.
+
+### Missing features
+Certain features like `flexWrap` on a `ASStackLayoutSpec` are not supported currently. Please see the section for `Layout Properties` for the full list of properties that are supported.
+
+
+## Layoutables
+To understand AsyncDisplayKit's layout system it's crucial to understand what a layoutable and a layout spec is. Layoutables are classes that conform to the `ASLayoutable` protocol. The `ASLayoutable` protocol declares a methods for measuring the layout of an object. A layout is defined by an `ASLayout` return value, and must specify
+  1. the size (but not position) of the layoutable object and
+  2. the size and position of all of its immediate child objects.
+
+`ASDisplayNode` as well as `ASLayoutSpec` classes are conforming to the `ASLayoutable` protocol and provide this functionally out of the box. This allows you to compose nodes and layout specs to build very complex layouts in a declarative way.
+
+### Replaced Layoutables
+A replaced layoutable is any layoutable whose appearance and dimensions are defined by an external resource. Examples include `ASImageNode`, `ASTextNode`, `ASButtonNode` or `ASTextNode`. All other layoutables can be referred to as non-replaced layoutables. This would include `ASVideoNode`, `ASVideoPlayerNode`, `ASNetworkImageNode` or `ASEditableTextNode`.
+
+Replaced layoutables can have intrinsic dimensions—width and height values that are defined by the layoutable itself, rather than by its surroundings. For example, if an `ASImageNode` has a width set to `ASDimensionAuto`, the width of the linked image file will be used. For an `ASImageNode` the intrinsic content size would be calculated based on the text content.
```

Comment:
I'm not sure if I truly understand "replaced layouts" based on these two paragraphs. It first says that it's a layoutable whose appearance and dimensions are defined by an external resource, then says that it can have intrinsic size that can be defined by itself rather than by its surroundings. I'm at odds on what I'm supposed to understand.

Could you discuss with me some examples that support these two cases and I think we'll be able to identify a simple, common definition that will be able to accurately describe the behavior of these types of layoutables.